Integrations/integration-jira.yml

```diff
@@ -58,30 +77,60 @@ configuration:
   name: fetchByCreated
   defaultvalue: ""
   type: 8
-  required: false  
+  required: false
 script:
   script: |-
     // handle '/' at the end of the url
     if (params.url[params.url.length - 1] != '/') {
         params.url = params.url + '/';
     }
 
-    var getCookies = function() {
-        var result = http(
-            params.url+'rest/auth/1/session',
-            {
-                Headers: {'Content-Type': ['application/json']},
-                Method: 'POST',
-                Username: params.username,
-                Password: params.password,
-                Body: JSON.stringify({'username':params.username,'password':params.password}),
-            },
-            params.insecure
-        );
+    var isOAuth;
+    if(!params.consumerKey || !params.accessToken || !params.privateKey) {
+        isOAuth = false;
+    } else {
+        isOAuth = true;
+    }
+
+    if(!isOAuth) {
+        if(!params.username ||!params.password) {
+            throw 'Please provide Authorization information, Basic or OAuth1.0'
+        }
+    }
+
+
+    var sendRequest = function (method, resourceUrl, body, isCookie) {
+        console.log("In send Request isOAuth: " + isOAuth  )
+        var requestParams = {
+            Headers: {'Content-Type': ['application/json']},
+            Method: method,
+        };
+        if(isOAuth) {
+            requestParams.OAuth1 = {
+                ConsumerKey:    params.consumerKey,
+                AccessToken:    params.accessToken,
+                PrivateKey:     params.privateKey
+            };
+            requestParams.Body = body !== undefined ? body : "";
+        } else {
+            requestParams.Username = params.username;
+            requestParams.Password = params.password;
+            requestParams.Body = body !== undefined ? body : "";
+            if(!isCookie) {
+                requestParams.Cookies = getCookies();
+            }
+        }
+
+        var result = http(params.url + resourceUrl, requestParams, params.insecure);
         if (result.StatusCode !== 200 && result.StatusCode !== 201) {
-            throw 'Failed to create auth cookie, request status code: ' + result.StatusCode + ', check that username/password are correct';
+            throw 'Failed to send request, request status code: ' + result.StatusCode + ', ' + result.Body;
         }
-        return result.Cookies;
+        return result;
```

Comment:
Dont you need to parse the result.body and return as object ?
Instead of doing it on all the callers